index.js

```diff
@@ -0,0 +1,400 @@
+    document.getElementById("input-hit").disabled = false;
+    document.getElementById("input-stand").disabled = false;
+
+    this.#playerStands = false;
+
+    this.state = GameState.Playing;
+
+    this.#currentDeck = new Deck();
+    this.#currentDeck.shuffle();
+
+    this.#players.forEach((player, index) => {
+      player.discardHand();
+
+      player.addCardToHand(this.#currentDeck.draw());
+      player.addCardToHand(this.#currentDeck.draw());
+
+      this.calculateHand(player);
+    });
+
+    const dealer = this.#players[this.#dealerIndex];
+    while (dealer.handValue < 17) {
+      dealer.addCardToHand(this.#currentDeck.draw());
+      this.calculateHand(dealer);
+    }
+
+    this.#players.forEach((player, index) => {
+      let delay = (index + 1) * 500;
+      player.getHand().forEach((card, cardIndex) => {
+        delay += (cardIndex + 1) * 500;
+        this.delay(delay).then(() => {
+          if (index === this.#dealerIndex) {
+            this.displayCard(card, DisplayArea.Dealer);
+          } else {
+            this.displayCard(card, DisplayArea.Player);
+          }
+
+          if (index === this.#dealerIndex) {
+            this.displayScore(player.handValue, DisplayArea.Dealer);
+          } else {
+            this.displayScore(player.handValue, DisplayArea.Player);
+          }
+        });
+      });
+    });
+
+    this.calculateScore();
+  }
+
+  calculateScore() {
+    const dealer = this.#players[this.#dealerIndex];
+    for (let i = 0; i < this.#players.length; i++) {
+      let player = this.#players[i];
+      if (i === this.#dealerIndex) {
+        if (player.handValue > 21) {
+          this.endTurn(true);
+          this.state = GameState.End;
+        } 
+      } else {
+        if (player.handValue > 21) {
+          this.endTurn(false);
+          this.state = GameState.End;
+        } else if (player.handValue === 21) {
+          if (dealer.handValue !== 21) {
+            this.endTurn(true);
+            this.state = GameState.End;
+          }
+        } else if (this.#playerStands) {
+          if (player.handValue > dealer.handValue) {
+            this.endTurn(true);
+            this.state = GameState.End;
+          } else if (player.handValue === dealer.handValue) {
+            this.endTurn(false);
+            player.addScore(this.#betAmount);
+            this.state = GameState.End;
+          } else {
+            this.endTurn(false);
+            this.state = GameState.End;
+          }
+        }
+      }
+    }
+  };
+
+  endTurn(playerWins) {
+    console.log("playerWins? " + playerWins);
+    const player = this.#players[0];
+
+    if (playerWins) {
+      player.addScore(this.#betAmount * this.#payoutRate);
+    }
+
+    document.getElementById("input-new-round").disabled = false;
+    document.getElementById("input-hit").disabled = true;
+    document.getElementById("input-stand").disabled = true;
+
+    this.displayBank(player);
+  }
```

Comment:
somehow endTurn(playerwins) could be better named, without reading the variable name in the function, i would have thought that 
endTurn(true) is equivalent to isTurnEnd=true
endTurn(false) is equivelent to isTurnEnd=false

could have thrown 
this.state = GameState.End;
in here too to reduce duplication of code above